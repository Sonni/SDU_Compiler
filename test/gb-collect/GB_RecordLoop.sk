var rec: record of {a: int, b: int, arr: array of int};
var arr: array of int;

allocate arr of length 4;
allocate rec;
allocate rec;

arr[1] = 42;

while true do
{
    allocate rec;
    write arr[1];
}

write 1;