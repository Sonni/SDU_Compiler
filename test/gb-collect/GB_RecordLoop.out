Should output 42 infinitely